examples/svm/plot_iris.py

```diff
@@ -39,55 +39,100 @@
 import matplotlib.pyplot as plt
 from sklearn import svm, datasets
 
+
+def make_meshgrid(x, y, h=.02):
+    """Create a mesh of points to plot in
+
+    Parameters
+    ----------
+    x: data to base x-axis meshgrid on
+    y: data to base y-axis meshgrid on
+    h: stepsize for meshgrid, optional
+
+    Returns
+    -------
+    xx, yy : ndarray
+    """
+    x_min, x_max = x.min() - 1, x.max() + 1
+    y_min, y_max = y.min() - 1, y.max() + 1
+    xx, yy = np.meshgrid(np.arange(x_min, x_max, h),
+                         np.arange(y_min, y_max, h))
+    return xx, yy
+
+
+def plot_contours(ax, clf, xx, yy, **params):
+    """Plot the decision boundaries for a classifier.
+
+    Parameters
+    ----------
+    ax: matplotlib axes object
+    clf: a classifier
+    xx: meshgrid ndarray
+    yy: meshgrid ndarray
+    params: dictionary of params to pass to contourf, optional
+    """
+    Z = clf.predict(np.c_[xx.ravel(), yy.ravel()])
+    Z = Z.reshape(xx.shape)
+    out = ax.contourf(xx, yy, Z, **params)
+    ax.set_xlim(xx.min(), xx.max())
+    ax.set_ylim(yy.min(), yy.max())
+    return out
+
+
+def plot_points_boundary(ax, clf, x, y, xx, yy,
+                         points_params, contour_params):
+    """Plot the decision boundaries and points.
+
+    Parameters
+    ----------
+    ax: matplotlib axes object
+    clf: a fitted classifier
+    x: x-axis of points to plot
+    y: y-axis of points to plot
+    xx: meshgrid ndarray
+    yy: meshgrid ndarray
+    point_params: dictionary of params to pass to `ax.scatter`, optional
+    countour_params: dictionary of params to pass to `plot_contours`, optional
+    """
+    contours = plot_contours(ax, clf, xx, yy, **contour_params)
```

Comment:
And I now see more clearly that the setting of `xlim` and `ylim` doesn't belong in the contour function.